domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/displaycontextparameter/DisplayContextParameterType.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.ccd.definition.store.domain.displaycontextparameter;
+
+import com.google.common.base.Strings;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public enum DisplayContextParameterType {
+    DATETIMEENTRY,
+    DATETIMEDISPLAY,
+    TABLE,
+    LIST;
+
+    private static final Pattern PATTERN = Pattern.compile("#(.+)\\((.+)\\)");
+    private static final int TYPE_GROUP = 1;
+    private static final int VALUE_GROUP = 2;
+
+    public static Optional<DisplayContextParameter> getDisplayContextParameterFor(String displayContextParameter) {
+        Optional<DisplayContextParameterType> type = getParameterTypeFor(displayContextParameter);
+        Optional<String> value = getParameterValueFor(displayContextParameter);
+
+        if (!type.isPresent() || !value.isPresent()) {
+            return Optional.empty();
+        }
+        return Optional.of(new DisplayContextParameter(type.get(), value.get()));
+    }
+
+    public static Optional<DisplayContextParameterType> getParameterTypeFor(String displayContextParameter) {
+        Matcher m = PATTERN.matcher(displayContextParameter);
+        if (m.matches()) {
+            try {
+                return Optional.of(valueOf(m.group(TYPE_GROUP)));
+            } catch (IllegalArgumentException e) {
+                return Optional.empty();
```

Comment:
mmm Optional.empty is usually used to model a lack of a value. Strictly speaking there is a value here, but it's invalid